backend/src-common/src/main/java/org/eclipse/sw360/datahandler/db/AttachmentAwareDatabaseHandler.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright Siemens AG, 2017. Part of the SW360 Portal Project.
+ *
+ * SPDX-License-Identifier: EPL-1.0
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+
+package org.eclipse.sw360.datahandler.db;
+
+import org.eclipse.sw360.datahandler.thrift.attachments.Attachment;
+import org.eclipse.sw360.datahandler.thrift.attachments.CheckStatus;
+
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import static org.eclipse.sw360.datahandler.common.CommonUtils.nullToEmptyCollection;
+import static org.eclipse.sw360.datahandler.common.CommonUtils.nullToEmptySet;
+
+public abstract class AttachmentAwareDatabaseHandler {
+
+    public void preventDeletionOfAcceptedAttachments(Set<Attachment> originalAttachments, Set<Attachment> changedAttachments) {
+        if(changedAttachments != null) {
```

Comment:
What happens if `changedAttachments` is null? It will allow deleting all attachments, including accepted ones? No bueno.
I think this method should return a `Set` that has to be used instead of `changedAttachments` by the caller. Then, the only thing you need to do here, is to get all accepted attachments from `originalAttachments` and add them to `changedAttachments`, then return that.